spec/oil-builtin-argparse.test.sh

```diff
@@ -2,34 +2,48 @@
 #
 # Some thoughts before writing code.  Hm can we do this entirely in user code, not as a builtin?
 #
-# I think it modifies OPT
+# The following is as close as possible to the python argparse which seems to work well
 
+#### Argparse boolsche option and positional
 
-#### Argparse Prototype
-
-hay define argparse
-
-# Oops, we're running into this problem ...
+#I assume the hay definitions are given.
+ArgSpec myspec {
+  arg -v --verbose { type = Bool }
+  # TODO: can position be implicit (in given order?) or is the order not maintained in hay?
```

Comment:
The order can be maintained (though we have to figure out how to do it)

Maybe add two positional args like `src` and `dest`?